app/templates/components/taxonomy-top-list.hbs

```diff
@@ -2,7 +2,7 @@
     <div class="subject row m-b-sm">
         {{#each pair as |subject|}}
             <div class="col-xs-12 col-sm-6 subject-item {{if (get subjectTooltips subject.text) 'hint--bottom hint--large'}}" aria-label={{get subjectTooltips subject.text}}>
-                {{#link-to (if theme.isProvider 'provider.discover' 'discover') (query-params subjectFilter=subject.text) class='btn btn-primary p-sm' invokeAction=(action "click" "link" (concat 'Preprints - Index - Browse ' subject.text))}}
+                {{#link-to (route-prefix 'discover') (query-params subjectFilter=subject.text) class='btn btn-primary p-sm' invokeAction=(action "click" "link" (concat 'Preprints - Index - Browse ' subject.text))}}
```

Comment:
This is a somewhat gnarly requirement. Have you taken a look at `Route#renderTemplate`? It seems to provide a way for one route to use the template or behavior of another. I'm not totally sure if it applies here, but might spark some ideas.

http://emberjs.com/api/classes/Ember.Route.html#method_renderTemplate